vm/src/eval.rs

```diff
@@ -277,17 +328,37 @@ pub fn eval_inst(vm: &mut NexusVM<impl Memory>) -> Result<()> {
         ECALL { rd } => {
             RD = rd;
             vm.Z = vm.syscalls.syscall(vm.regs.pc, vm.regs.x, &vm.mem)?;
+            // Profile cycles
+            if vm.Z == SyscallCode::ProfileCycles as u32 {
+                handle_profile_cycles(vm)?;
+            }
         }
         UNIMP => {
             PC = vm.inst.pc;
         }
     }
 
+    // Counts cycles per instruction kind.
+    // In RISC-V:
+    // - Memory instructions are 3 cycles
+    // - Branch instructions are 2 cycles
+    // - System call instruction are 4 cycles
```

Comment:
Minor nit but would be nice for the ordering on the comment to match the ordering of the pattern matching.

Reply:
updated in https://github.com/nexus-xyz/nexus-zkvm/pull/243/commits/0cf58534f504fb1bbf5a475e0da2d35ed1a05916